transport/http/outbound.go

```diff
@@ -90,6 +94,24 @@ func AddHeader(key, value string) OutboundOption {
 	}
 }
 
+// WithRequestFactory allows end clients to override the default behavior for creating HTTP requests
+// from the underlying transport request. This is useful if the remote service doesn't support
+// YARPC transport headers:
+//
+//  httpTransport.NewOutbound(chooser, http.WithRequestFactory(
+//    func(treq *transport.Request) *http.Request {
+//      newURL, _ := url.Parse(fmt.Sprintf("http://host:port/%s", treq.Procedure))
+//      return http.NewRequest("POST", newURL.String(), treq.Body)
+//  })
+//
+// An example would be an API gateway or mapping RPC methods to a legacy service. When overriding
+// the URL, do not modify the base URI host or port.
```

Comment:
I think this is good advice but I don't fully comprehend what YARPC internals try to do - if you use this option to violate the peer identifier assumption I'm not sure what the consequences will be.